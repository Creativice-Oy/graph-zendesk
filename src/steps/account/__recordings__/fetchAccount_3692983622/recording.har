{
  "log": {
    "_recordingName": "fetchAccount",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "6963c6f3f589a3e055a06cafb0ba8004",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "jupiterone-dev.zendesk.com"
            }
          ],
          "headersSize": 319,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://jupiterone-dev.zendesk.com/api/v2/account"
        },
        "response": {
          "bodySize": 362,
          "content": {
            "_isBinary": true,
            "mimeType": "application/json; charset=UTF-8",
            "size": 362,
            "text": "[\"1f8b080000000000000355cdbd0e82401004e077d91a811090789dad85d656e4b85be209774bee470c847777136d2c67f2656603a9142517416c90fc04021e31ce4114c533cd26a22787078daf7c45a7318cb9220b19386991ede56b6e0eb90bd2e99ede20063905e49c7a4d561ac7f07f8c713416bb81bc957c5dd5bf62e537d6678bde28595c71e9eee447f6b438f49dd120eaaa6c9be6d8b6e5a9cac0a6299ad9934e8a87a24fb8ef1f4a49d726d5000000\"]"
          },
          "cookies": [
            {
              "domain": ".jupiterone-dev.zendesk.com",
              "httpOnly": true,
              "name": "__cfruid",
              "path": "/",
              "sameSite": "None",
              "secure": true,
              "value": "[REDACTED]"
            }
          ],
          "headers": [
            {
              "name": "date",
              "value": "Fri, 28 Jan 2022 14:12:26 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "transfer-encoding",
              "value": "chunked"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "zendesk-api-version",
              "value": "2022-01-01"
            },
            {
              "name": "x-zendesk-api-version",
              "value": "v2"
            },
            {
              "name": "x-zendesk-application-version",
              "value": "v10911"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "access-control-allow-origin",
              "value": "*"
            },
            {
              "name": "access-control-allow-credentials",
              "value": "true"
            },
            {
              "name": "access-control-expose-headers",
              "value": "X-Zendesk-API-Warn,X-Zendesk-User-Id,X-Zendesk-User-Session-Expires-At"
            },
            {
              "name": "x-rate-limit",
              "value": "200"
            },
            {
              "name": "x-rate-limit-remaining",
              "value": "198"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000;"
            },
            {
              "name": "etag",
              "value": "W/\"a80e3e81a54a847723877a9cbb1366d9\""
            },
            {
              "name": "cache-control",
              "value": "max-age=0, private, must-revalidate"
            },
            {
              "name": "x-zendesk-origin-server",
              "value": "classic-app-server-7f77698bc-76jkn"
            },
            {
              "name": "x-request-id",
              "value": "6d4ad0d01f992cad-SEA, 6d4ad0d01f992cad-SEA"
            },
            {
              "name": "x-runtime",
              "value": "0.132651"
            },
            {
              "name": "x-zendesk-zorg",
              "value": "yes"
            },
            {
              "name": "content-encoding",
              "value": "gzip"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "cf-cache-status",
              "value": "DYNAMIC"
            },
            {
              "name": "expect-ct",
              "value": "max-age=604800, report-uri=\"https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct\""
            },
            {
              "name": "report-to",
              "value": "{\"endpoints\":[{\"url\":\"https:\\/\\/a.nel.cloudflare.com\\/report\\/v3?s=KKvKXnE9Xaq%2BdCULl657hTV8di593N3D5E6TIL26%2BIjm2F6vX2Lwy2SYkCJ2H%2BHJ9tv2al0n%2F0BKpRJPn6ma2XT4iugDHAafl%2FvgRlWuwXJy09puzn1zYZcECa0VB%2FH%2BYtFBy%2FomY%2FKR9m3R\"}],\"group\":\"cf-nel\",\"max_age\":604800}"
            },
            {
              "name": "nel",
              "value": "{\"success_fraction\":0.01,\"report_to\":\"cf-nel\",\"max_age\":604800}"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "name": "server",
              "value": "cloudflare"
            },
            {
              "name": "cf-ray",
              "value": "6d4ad0d01f992cad-ORD"
            }
          ],
          "headersSize": 1521,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2022-01-28T14:12:26.089Z",
        "time": 372,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 372
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
