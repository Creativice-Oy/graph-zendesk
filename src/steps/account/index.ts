import {
  IntegrationStep,
  IntegrationStepExecutionContext,
} from '@jupiterone/integration-sdk-core';
import { createAPIClient } from '../../client';
import { IntegrationConfig } from '../../config';
import { createAccountEntity } from './converters';
import { ACCOUNT_ENTITY_KEY, Entities, IntegrationSteps } from '../constants';

// Oauth scope: 'read'
export async function fetchAccount({
  instance,
  jobState,
}: IntegrationStepExecutionContext<IntegrationConfig>) {
  const apiClient = createAPIClient(instance.config);

  const currentUser = await apiClient.getCurrentUser();
  const accountEntity = createAccountEntity(currentUser);
  await jobState.addEntity(accountEntity);
  await jobState.setData(ACCOUNT_ENTITY_KEY, accountEntity);
}

export const accountSteps: IntegrationStep<IntegrationConfig>[] = [
  {
    id: IntegrationSteps.ACCOUNT,
    name: 'Fetch Account',
    entities: [Entities.ACCOUNT],
    relationships: [],
    dependsOn: [],
    executionHandler: fetchAccount,
  },
];
