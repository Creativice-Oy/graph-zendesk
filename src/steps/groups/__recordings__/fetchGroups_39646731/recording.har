{
  "log": {
    "_recordingName": "fetchGroups",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "ed2fd5f5d5cd127ce3120f22e7ddec13",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "jupiterone-dev.zendesk.com"
            }
          ],
          "headersSize": 330,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "page",
              "value": "1"
            }
          ],
          "url": "https://jupiterone-dev.zendesk.com/api/v2/groups.json?page=1"
        },
        "response": {
          "bodySize": 426,
          "content": {
            "_isBinary": true,
            "mimeType": "application/json; charset=UTF-8",
            "size": 426,
            "text": "[\"1f8b0800000000000003858ecd6e02310c84dfc5e76593cd56fdc96bc0098450b46b686848acc45e5545bc7bddf6524e48becc68e6f35ce1548b5003bfbb82d4041ede99a97963ce4291b1968cab1997fe0bf38ceda39fcac5048a6671e6af6ac6676bddf032beba71e8cfad64e820cee0fffb1de47041a5af85a854d688d2a61a89a3163cfc1ac72089c17315fc91091995730ca9a99e2a06d587a00970fa713558bd8db3fee9cd8f76ab08a1f941e6b6d729f8c9070a27dd9325a50ea8e2128bb43b732a92f5d570fb0619c2c7d924010000\"]"
          },
          "cookies": [
            {
              "domain": ".jupiterone-dev.zendesk.com",
              "httpOnly": true,
              "name": "__cfruid",
              "path": "/",
              "sameSite": "None",
              "secure": true,
              "value": "[REDACTED]"
            }
          ],
          "headers": [
            {
              "name": "date",
              "value": "Tue, 16 Nov 2021 16:13:38 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "transfer-encoding",
              "value": "chunked"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "x-zendesk-api-version",
              "value": "v2"
            },
            {
              "name": "x-zendesk-application-version",
              "value": "v10370"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "access-control-allow-origin",
              "value": "*"
            },
            {
              "name": "access-control-allow-credentials",
              "value": "true"
            },
            {
              "name": "access-control-expose-headers",
              "value": "X-Zendesk-API-Warn,X-Zendesk-User-Id,X-Zendesk-User-Session-Expires-At"
            },
            {
              "name": "x-rate-limit",
              "value": "200"
            },
            {
              "name": "x-rate-limit-remaining",
              "value": "198"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000;"
            },
            {
              "name": "etag",
              "value": "W/\"76e3a2fd91ec6717583c8b3cf6682ee8\""
            },
            {
              "name": "cache-control",
              "value": "max-age=0, private, must-revalidate"
            },
            {
              "name": "x-zendesk-origin-server",
              "value": "classic-app-server-7dd6688cff-bh6lh"
            },
            {
              "name": "x-request-id",
              "value": "6af201f8f9c27c64-SEA, 6af201f8f9c27c64-SEA"
            },
            {
              "name": "x-runtime",
              "value": "0.120527"
            },
            {
              "name": "x-zendesk-zorg",
              "value": "yes"
            },
            {
              "name": "content-encoding",
              "value": "gzip"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "cf-cache-status",
              "value": "DYNAMIC"
            },
            {
              "name": "expect-ct",
              "value": "max-age=604800, report-uri=\"https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct\""
            },
            {
              "name": "report-to",
              "value": "{\"endpoints\":[{\"url\":\"https:\\/\\/a.nel.cloudflare.com\\/report\\/v3?s=ftz1mV4ZaT%2BxFkdrZX8VwXAL1rkiyopvk6ugPMBZ9RG5GKDMHsyr%2Bvo7GXWYmAKgYQ2qdf142AFfYpDZmlBAnf%2FQyL0UiE0XQ5nbZibuNc7cs3zq16SnFZGo9pB83MKfGS8tLmzDWwWbAKhA\"}],\"group\":\"cf-nel\",\"max_age\":604800}"
            },
            {
              "name": "nel",
              "value": "{\"success_fraction\":0.01,\"report_to\":\"cf-nel\",\"max_age\":604800}"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "name": "server",
              "value": "cloudflare"
            },
            {
              "name": "cf-ray",
              "value": "6af201f8f9c27c64-BEG"
            }
          ],
          "headersSize": 1477,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-11-16T16:13:38.038Z",
        "time": 391,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 391
        }
      },
      {
        "_id": "6963c6f3f589a3e055a06cafb0ba8004",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "jupiterone-dev.zendesk.com"
            }
          ],
          "headersSize": 319,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://jupiterone-dev.zendesk.com/api/v2/account"
        },
        "response": {
          "bodySize": 362,
          "content": {
            "_isBinary": true,
            "mimeType": "application/json; charset=UTF-8",
            "size": 362,
            "text": "[\"1f8b080000000000000355cdbd0e82401004e077d91a811090789dad85d656e4b85be209774bee470c847777136d2c67f2656603a9142517416c90fc04021e31ce4114c533cd26a22787078daf7c45a7318cb9220b19386991ede56b6e0eb90bd2e99ede20063905e49c7a4d561ac7f07f8c713416bb81bc957c5dd5bf62e537d6678bde28595c71e9eee447f6b438f49dd120eaaa6c9be6d8b6e5a9cac0a6299ad9934e8a87a24fb8ef1f4a49d726d5000000\"]"
          },
          "cookies": [
            {
              "domain": ".jupiterone-dev.zendesk.com",
              "httpOnly": true,
              "name": "__cfruid",
              "path": "/",
              "sameSite": "None",
              "secure": true,
              "value": "[REDACTED]"
            }
          ],
          "headers": [
            {
              "name": "date",
              "value": "Fri, 28 Jan 2022 14:12:26 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "transfer-encoding",
              "value": "chunked"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "zendesk-api-version",
              "value": "2022-01-01"
            },
            {
              "name": "x-zendesk-api-version",
              "value": "v2"
            },
            {
              "name": "x-zendesk-application-version",
              "value": "v10911"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "access-control-allow-origin",
              "value": "*"
            },
            {
              "name": "access-control-allow-credentials",
              "value": "true"
            },
            {
              "name": "access-control-expose-headers",
              "value": "X-Zendesk-API-Warn,X-Zendesk-User-Id,X-Zendesk-User-Session-Expires-At"
            },
            {
              "name": "x-rate-limit",
              "value": "200"
            },
            {
              "name": "x-rate-limit-remaining",
              "value": "198"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000;"
            },
            {
              "name": "etag",
              "value": "W/\"a80e3e81a54a847723877a9cbb1366d9\""
            },
            {
              "name": "cache-control",
              "value": "max-age=0, private, must-revalidate"
            },
            {
              "name": "x-zendesk-origin-server",
              "value": "classic-app-server-7f77698bc-gddwc"
            },
            {
              "name": "x-request-id",
              "value": "6d4ad0d0195f2c1d-SEA, 6d4ad0d0195f2c1d-SEA"
            },
            {
              "name": "x-runtime",
              "value": "0.112813"
            },
            {
              "name": "x-zendesk-zorg",
              "value": "yes"
            },
            {
              "name": "content-encoding",
              "value": "gzip"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "cf-cache-status",
              "value": "DYNAMIC"
            },
            {
              "name": "expect-ct",
              "value": "max-age=604800, report-uri=\"https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct\""
            },
            {
              "name": "report-to",
              "value": "{\"endpoints\":[{\"url\":\"https:\\/\\/a.nel.cloudflare.com\\/report\\/v3?s=EdlTAtYyfttlxjZxwPRro7tUsm42UrUZpgqyx32Awq00xot0U4bVar0igesX9IA%2BZQetvgDj5FWf1UHyfA4pbwLTLDGQGvGmLHqfTT51KmrWtUh8DITK%2FkiIlWML4EViiAzIvxTNNe4V1uav\"}],\"group\":\"cf-nel\",\"max_age\":604800}"
            },
            {
              "name": "nel",
              "value": "{\"success_fraction\":0.01,\"report_to\":\"cf-nel\",\"max_age\":604800}"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "name": "server",
              "value": "cloudflare"
            },
            {
              "name": "cf-ray",
              "value": "6d4ad0d0195f2c1d-ORD"
            }
          ],
          "headersSize": 1507,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2022-01-28T14:12:26.091Z",
        "time": 370,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 370
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
