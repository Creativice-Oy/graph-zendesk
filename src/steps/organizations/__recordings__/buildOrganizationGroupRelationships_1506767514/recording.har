{
  "log": {
    "_recordingName": "buildOrganizationGroupRelationships",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "5.1.1"
    },
    "entries": [
      {
        "_id": "ed2fd5f5d5cd127ce3120f22e7ddec13",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "jupiterone-dev.zendesk.com"
            }
          ],
          "headersSize": 330,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "page",
              "value": "1"
            }
          ],
          "url": "https://jupiterone-dev.zendesk.com/api/v2/groups.json?page=1"
        },
        "response": {
          "bodySize": 426,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 426,
            "text": "{\"groups\":[{\"url\":\"https://jupiterone-dev.zendesk.com/api/v2/groups/360021738231.json\",\"id\":360021738231,\"name\":\"Support\",\"description\":\"\",\"default\":true,\"deleted\":false,\"created_at\":\"2021-10-10T20:49:30Z\",\"updated_at\":\"2021-10-10T20:49:30Z\"}],\"next_page\":null,\"previous_page\":null,\"count\":1}"
          },
          "cookies": [
            {
              "domain": ".jupiterone-dev.zendesk.com",
              "httpOnly": true,
              "name": "__cfruid",
              "path": "/",
              "sameSite": "None",
              "secure": true,
              "value": "[REDACTED]"
            }
          ],
          "headers": [
            {
              "name": "date",
              "value": "Fri, 28 Jan 2022 14:53:11 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "zendesk-api-version",
              "value": "2022-01-01"
            },
            {
              "name": "x-zendesk-api-version",
              "value": "v2"
            },
            {
              "name": "x-zendesk-application-version",
              "value": "v10911"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "access-control-allow-origin",
              "value": "*"
            },
            {
              "name": "access-control-allow-credentials",
              "value": "true"
            },
            {
              "name": "access-control-expose-headers",
              "value": "X-Zendesk-API-Warn,X-Zendesk-User-Id,X-Zendesk-User-Session-Expires-At"
            },
            {
              "name": "x-rate-limit",
              "value": "200"
            },
            {
              "name": "x-rate-limit-remaining",
              "value": "188"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000;"
            },
            {
              "name": "etag",
              "value": "W/\"76e3a2fd91ec6717583c8b3cf6682ee8\""
            },
            {
              "name": "cache-control",
              "value": "max-age=0, private, must-revalidate"
            },
            {
              "name": "x-zendesk-origin-server",
              "value": "classic-app-server-7f77698bc-dhhzn"
            },
            {
              "name": "x-request-id",
              "value": "6d4b0c8258682c9b-SEA, 6d4b0c8258682c9b-SEA"
            },
            {
              "name": "x-runtime",
              "value": "0.157789"
            },
            {
              "name": "x-zendesk-zorg",
              "value": "yes"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "cf-cache-status",
              "value": "DYNAMIC"
            },
            {
              "name": "expect-ct",
              "value": "max-age=604800, report-uri=\"https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct\""
            },
            {
              "name": "report-to",
              "value": "{\"endpoints\":[{\"url\":\"https:\\/\\/a.nel.cloudflare.com\\/report\\/v3?s=XPTIwuN92VUgnkukFKMtQDBHcQzyLBrZgV5XahDd4%2F%2FjMHDyHr6ig%2FNeYTX7%2B6p%2Bx%2FOWzfWVFSYGhfHGLym%2F69SYCq%2FAyYDdiZM3sIEO5OpGCDHQmMivZ%2FofPjthAr%2FZdrvxDABzdtrE3TdQ\"}],\"group\":\"cf-nel\",\"max_age\":604800}"
            },
            {
              "name": "nel",
              "value": "{\"success_fraction\":0.01,\"report_to\":\"cf-nel\",\"max_age\":604800}"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "name": "server",
              "value": "cloudflare"
            },
            {
              "name": "cf-ray",
              "value": "6d4b0c8258682c9b-ORD"
            }
          ],
          "headersSize": 1523,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2022-01-28T14:53:11.248Z",
        "time": 330,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 330
        }
      },
      {
        "_id": "de45a5ef1f284a357f682f3933d50fc4",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "jupiterone-dev.zendesk.com"
            }
          ],
          "headersSize": 337,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "page",
              "value": "1"
            }
          ],
          "url": "https://jupiterone-dev.zendesk.com/api/v2/organizations.json?page=1"
        },
        "response": {
          "bodySize": 574,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 574,
            "text": "{\"organizations\":[{\"url\":\"https://jupiterone-dev.zendesk.com/api/v2/organizations/361584776731.json\",\"id\":361584776731,\"name\":\"JupiterOne\",\"shared_tickets\":false,\"shared_comments\":false,\"external_id\":null,\"created_at\":\"2021-10-10T20:49:30Z\",\"updated_at\":\"2021-11-05T16:50:42Z\",\"domain_names\":[],\"details\":\"\",\"notes\":\"\",\"group_id\":360021738231,\"tags\":[],\"organization_fields\":{}}],\"next_page\":null,\"previous_page\":null,\"count\":1}"
          },
          "cookies": [
            {
              "domain": ".jupiterone-dev.zendesk.com",
              "httpOnly": true,
              "name": "__cfruid",
              "path": "/",
              "sameSite": "None",
              "secure": true,
              "value": "[REDACTED]"
            }
          ],
          "headers": [
            {
              "name": "date",
              "value": "Fri, 28 Jan 2022 14:53:12 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "zendesk-api-version",
              "value": "2022-01-01"
            },
            {
              "name": "x-zendesk-api-version",
              "value": "v2"
            },
            {
              "name": "x-zendesk-application-version",
              "value": "v10911"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "etag",
              "value": "W/\"aee7c5535e2c5f8d50b0f209ef410549\""
            },
            {
              "name": "access-control-allow-origin",
              "value": "*"
            },
            {
              "name": "access-control-allow-credentials",
              "value": "true"
            },
            {
              "name": "access-control-expose-headers",
              "value": "X-Zendesk-API-Warn,X-Zendesk-User-Id,X-Zendesk-User-Session-Expires-At"
            },
            {
              "name": "cache-control",
              "value": "max-age=30, private"
            },
            {
              "name": "x-rate-limit",
              "value": "200"
            },
            {
              "name": "x-rate-limit-remaining",
              "value": "186"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000;"
            },
            {
              "name": "x-zendesk-origin-server",
              "value": "classic-app-server-7f77698bc-h8qwx"
            },
            {
              "name": "x-request-id",
              "value": "6d4b0c847f38637e-SEA, 6d4b0c847f38637e-SEA"
            },
            {
              "name": "x-runtime",
              "value": "0.170101"
            },
            {
              "name": "x-zendesk-zorg",
              "value": "yes"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "cf-cache-status",
              "value": "DYNAMIC"
            },
            {
              "name": "expect-ct",
              "value": "max-age=604800, report-uri=\"https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct\""
            },
            {
              "name": "report-to",
              "value": "{\"endpoints\":[{\"url\":\"https:\\/\\/a.nel.cloudflare.com\\/report\\/v3?s=cwSOtMOdmMFOmm0wOfCOVybxLpR2lnocQoKrW%2ByKdu4FpPu79HPRmApyv6wVvHKH9hg8fzNOY0HpbLD19c7oJt2ct%2F5yIjQx3OfZJ4p2ccwU8xX8Wb%2BPgrvBUpbB6fLPkrtsRPF9egYdEoRS\"}],\"group\":\"cf-nel\",\"max_age\":604800}"
            },
            {
              "name": "nel",
              "value": "{\"success_fraction\":0.01,\"report_to\":\"cf-nel\",\"max_age\":604800}"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "name": "server",
              "value": "cloudflare"
            },
            {
              "name": "cf-ray",
              "value": "6d4b0c847f38637e-ORD"
            }
          ],
          "headersSize": 1493,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2022-01-28T14:53:11.581Z",
        "time": 365,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 365
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
