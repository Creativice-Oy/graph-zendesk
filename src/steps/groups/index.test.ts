import {
  createMockStepExecutionContext,
  Recording,
} from '@jupiterone/integration-sdk-testing';
import { fetchGroups } from '.';
import { integrationConfig } from '../../../test/config';
import { setupZendeskRecording } from '../../../test/recording';
import { IntegrationConfig } from '../../config';
import { fetchAccount } from '../account';
import { Entities } from '../constants';

describe('#fetchGroups', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupZendeskRecording({
      directory: __dirname,
      name: 'fetchGroups',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: integrationConfig,
    });

    await fetchAccount(context);
    await fetchGroups(context);

    const groups = context.jobState.collectedEntities.filter((e) =>
      e._type.includes(Entities.GROUP._type),
    );

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      numCollectedRelationships: context.jobState.collectedRelationships.length,
      collectedEntities: context.jobState.collectedEntities,
      collectedRelationships: context.jobState.collectedRelationships,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    expect(groups.length).toBeGreaterThan(0);
    expect(groups).toMatchGraphObjectSchema({
      _class: Entities.GROUP._class,
      schema: {
        additionalProperties: false,
        required: [],
        properties: {
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          _type: { const: Entities.GROUP._type },
          _class: { const: Entities.GROUP._class },
          _key: { type: 'string' },
          createdAt: { type: 'string' },
          deleted: { type: 'boolean' },
          description: { type: 'string' },
          id: { type: 'string' },
          name: { type: 'string' },
          updatedAt: { type: 'string' },
          url: { type: 'string' },
        },
      },
    });
  });
});
