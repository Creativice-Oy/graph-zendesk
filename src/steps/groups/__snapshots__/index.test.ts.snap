// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#fetchGroups should collect data 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "Account",
      ],
      "_key": "zendesk_account:jupiterone-dev",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "multiproduct": true,
            "name": "JupiterOne",
            "owner_id": 420755677092,
            "sandbox": false,
            "subdomain": "jupiterone-dev",
            "time_format": 24,
            "time_zone": "America/New_York",
            "url": "https://jupiterone-dev.zendesk.com",
          },
        },
      ],
      "_type": "zendesk_account",
      "accessURL": "https://jupiterone-dev.zendesk.com",
      "accountId": "jupiterone-dev",
      "createdOn": undefined,
      "displayName": "JupiterOne",
      "name": "JupiterOne",
    },
    Object {
      "_class": Array [
        "Group",
      ],
      "_key": "zendesk_group:360021738231",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "created_at": "2021-10-10T20:49:30Z",
            "default": true,
            "deleted": false,
            "description": "",
            "id": 360021738231,
            "name": "Support",
            "updated_at": "2021-10-10T20:49:30Z",
            "url": "https://jupiterone-dev.zendesk.com/api/v2/groups/360021738231.json",
          },
        },
      ],
      "_type": "zendesk_group",
      "createdAt": "2021-10-10T20:49:30Z",
      "createdOn": undefined,
      "deleted": false,
      "description": "",
      "displayName": "Support",
      "id": "360021738231",
      "name": "Support",
      "updatedAt": "2021-10-10T20:49:30Z",
      "url": "https://jupiterone-dev.zendesk.com/api/v2/groups/360021738231.json",
    },
  ],
  "collectedRelationships": Array [],
  "encounteredTypes": Array [
    "zendesk_account",
    "zendesk_group",
  ],
  "numCollectedEntities": 2,
  "numCollectedRelationships": 0,
}
`;
